packages/odf/components/s3-browser/upload-objects/utils.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  humanizeBinaryBytes,
+  humanizeDecimalBytesPerSec,
+  humanizeMinutes,
+  humanizeSeconds,
+} from '@odf/shared/utils';
+import * as _ from 'lodash-es';
+import { UploadProgress, UploadStatus } from './types';
+
+export const getCompletedAndTotalUploadObjects = (objects: UploadProgress) => {
+  const totalObjects = Object.keys(objects).length;
+  const totalUploaded = Object.keys(objects).reduce((acc, curr) => {
+    if (objects[curr].total === objects[curr].loaded) {
+      acc += 1;
+      return acc;
+    }
+    return acc;
+  }, 0);
+  return [totalUploaded, totalObjects];
+};
+
+export const getUploadSpeed = (objects: UploadProgress): string => {
+  if (_.isEmpty(objects)) return '';
+  const files = Object.values(objects);
+  const uploadingFile = files.filter(
+    (item) => item.uploadState === UploadStatus.UPLOAD_START
+  )?.[0];
+  const totalTime = Date.now() - (uploadingFile?.startTime || Date.now());
+  const totalUploaded = uploadingFile?.loaded ?? 0;
+  const speed = humanizeDecimalBytesPerSec(totalUploaded / (totalTime / 1000));
+  return speed.string;
+};
+
+export const getTotalRemainingFilesAndSize = (
+  objects: UploadProgress
+): string => {
+  if (_.isEmpty(objects)) return '';
+  const files = Object.values(objects);
+  const uploadingFiles = files.filter(
+    (item) => item.uploadState === UploadStatus.UPLOAD_START
+  );
+  const filesCount = uploadingFiles.length;
+  const filesSize = uploadingFiles.reduce(
+    (acc, curr) => (acc += curr.total - curr.loaded),
+    0
+  );
+  return `${filesCount} files (${humanizeBinaryBytes(filesSize).string})`;
+};
+
+export const getTotalTimeReamining = (objects: UploadProgress): string => {
```

Comment:
same as https://github.com/red-hat-storage/odf-console/pull/1635/files#r1795811268
```suggestion
export const getTotalUploadTimeRemaining = (objects: UploadProgress): string => {
```

Reply:
or if we wanna call it `getTotalTimeReamining `, then include `INIT_STATE` as well ??